tests/virt_autotest/esxi_open_vm_tools.pm

```diff
@@ -0,0 +1,217 @@
+
+    record_info('Guest Power Shutdown');
+    check_vm_power_state($vm_id, $vm_ip, 'power.shutdown', 0, 1, $VM_POWERED_ON, $VM_POWERED_OFF);
+}
+
+sub do_networking_tests {
+    my ($vm_id, $vm_ip) = @_;
+
+    record_info('Networking Tests');
+    check_vm_power_state($vm_id, $vm_ip, 'power.on', 1, 0, $VM_POWERED_OFF, $VM_POWERED_ON);
+
+    my $vswitch_name = esxi_vm_network_binding($vm_id);
+    record_info('VSwitch Name', $vswitch_name);
+
+    login_vm_console();
+
+    assert_script_run("ping -I $vm_ip -4 -c3 openqa.suse.de");
+    assert_script_run("ping -I $vm_ip -4 -c3 www.suse.com");
+}
+
+sub do_clock_sync_tests {
+    my ($vm_name, $vm_id, $vm_ip) = @_;
+    my ($g_datetime, $h_datetime, $g_difftime);
+
+    record_info('Clock Sync Tests');
+    check_vm_power_state($vm_id, $vm_ip, 'power.on', 1, 0, $VM_POWERED_OFF, $VM_POWERED_ON);
+    select_console('sut') if (current_console() ne 'sut');
+
+    # Set the last day of host time within guest before enabling timesynce
+    $g_datetime = set_random_time();
+    record_info('Guest current time before syncing up', $g_datetime);
+
+    # Enable timesync service in guest VM
+    if (script_output('vmware-toolbox-cmd timesync status', proceed_on_failure => 1, timeout => 90) eq 'Disabled') {
+        assert_script_run('vmware-toolbox-cmd timesync enable', timeout => 90);
+        $h_datetime = get_host_timestamp();
+        record_info('Host time', $h_datetime);
+        $g_datetime = script_output("date -u +'\%F \%T'");
+        record_info('Guest time after syncing up with host', $g_datetime);
+    }
+
+    # Check the time difference between host and guest
+    my $h_timesec = get_epoch_time($h_datetime);
+    my $g_timesec = get_epoch_time($g_datetime);
+    my $diff_secs = int($g_timesec - $h_timesec);
+
+    record_info('Diff secs', $diff_secs);
+    if ($diff_secs < 10) {
+        record_info('Clock synchronization is successful.');
+    } else {
+        die 'Clock synchronization failed.';
+    }
+
+    # Disable clock sync tests
+    select_console('svirt');
+    check_vm_power_state($vm_id, $vm_ip, 'power.shutdown', 0, 0, $VM_POWERED_ON, $VM_POWERED_OFF);
+    disable_vm_time_synchronization($vm_name);
+    check_vm_power_state($vm_id, $vm_ip, 'power.on', 1, 0, $VM_POWERED_OFF, $VM_POWERED_ON);
+    login_vm_console();
+
+    # Set the guest system time which different from the host
+    $g_difftime = set_random_time();
+    record_info('Set a time in guest that different from host', $g_difftime);
+
+    # Guest time will not be synced up with the hypervisor after time synchronization disabled
+    $g_datetime = script_output("date -u +'\%F \%T'");
+    $h_datetime = get_host_timestamp();
+    $g_timesec  = get_epoch_time($g_datetime);
+    $h_timesec  = get_epoch_time($h_datetime);
+    $diff_secs  = int($g_timesec - $h_timesec);
+
+    record_info('Diff secs', $diff_secs);
+    if (abs($diff_secs) > 300) {
```

Comment:
The guest time is set randomly, although chance is low, but in theory it is possible to be within 300 seconds difference. Suggest to set guest time with given time difference with host time, eg host time -1 day.

Besides, in this do_clock_sync_tests, there are obvious duplicated code lines which can be saved by small function, eg compare time difference code block. Can you improve it?

Reply:
Also you checking the time difference twice - I suggest to create a subroutine for this.

Reply:
Updated to set guest time with the last day of host time, and creating a subroutine "init_guest_time()" for both the case of  timesync enable and disable. And created another subroutine "get_diff_seconds()" for checking the time difference code block.

Reply:
Thanks for the rework. It well matches the suggestion comments . 

One more minor suggestion, with 1 day/3600 seconds difference, shall the diff check here be closer to 3600 seconds, so as to ensure that the setting and syncing disablement actually works?

Reply:
Here the guest time is set one day earlier than host, the purpose of this test is to ensure the time will not be synced with host, but it's not able to avoid any time drift from guest. As we know, some intermediate operations(e.g. getting timestamp from host&guest) will cost a few seconds. I think as long as we observe the time is out of sync, that's enough.

Reply:
OK, let's take an example. Refer to the verification run - http://10.67.129.66/tests/351#step/esxi_open_vm_tools/224

After timesync disabled and reset the guest time, we can see the deviation should be within 10s. I think it is acceptable.
Host time: 2021-09-26 07:21:50
Guest time: 2021-09-25 07:21:40

Reply:
Got it. Updated.